tst/com/amazon/corretto/crypto/provider/test/EvpKeyFactoryTest.java

```diff
@@ -86,17 +86,10 @@ public static void setupParameters() throws Exception {
     }
 
     for (String algorithm : ALGORITHMS) {
-      KeyPairGenerator kpg;
-      if (algorithm.startsWith("ML-DSA")
-          || (algorithm.startsWith("Ed") && TestUtil.getJavaVersion() < 15)) {
-        // JCE doesn't support ML-DSA until JDK24, and BouncyCastle currently
-        // serializes ML-DSA private keys via seeds.
-        // TODO: switch to BouncyCastle once BC supports CHOICE-encoded private keys
-        // Similarly, JDK doesn't support EdDSA/Ed25519 until JDK15
-        kpg = KeyPairGenerator.getInstance(algorithm, NATIVE_PROVIDER);
-      } else {
-        kpg = KeyPairGenerator.getInstance(algorithm);
-      }
+      KeyPairGenerator kpg =
+          getAlternateProvider(algorithm) == null
+              ? KeyPairGenerator.getInstance(algorithm)
+              : KeyPairGenerator.getInstance(algorithm, getAlternateProvider(algorithm));
```

Comment:
NP: Instead of repeating the ternary operator everywhere this is called, just have the function return the value that's needed. (Of course, the function name would also need to be changed.)

Reply:
Ahh.. it's `KeyPairGenerator` here and a `KeyFactory` for the other two.